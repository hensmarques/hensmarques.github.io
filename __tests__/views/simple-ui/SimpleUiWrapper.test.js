import React from 'react';
import { SimpleUiWrapper } from '../../../src/views/simple-ui/SimpleUiWrapper';

describe('<SimpleUiWrapper/>', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<SimpleUiWrapper title="mock title" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the props and children correctly', () => {
    const translationSpy = jest.spyOn(window, '_t').mockClear();
    const mockProps = {
      description: 'mock description',
      pageHeader: 'mock page header',
      translation: 'en',
    };
    const wrapper = shallow(
      <SimpleUiWrapper {...mockProps}>
        <p>mock child component</p>
      </SimpleUiWrapper>,
    );
    expect(wrapper.containsMatchingElement(<h2>{mockProps.pageHeader}</h2>)).toBe(true);
    expect(wrapper.containsMatchingElement(<p>{mockProps.description}</p>)).toBe(true);
    expect(wrapper.containsMatchingElement(<p>mock child component</p>)).toBe(true);

    expect(translationSpy).toHaveBeenCalledTimes(2);
    expect(translationSpy).toHaveBeenCalledWith(
      mockProps.pageHeader,
      `${mockProps.translation}.PAGE_HEADER`,
    );
    expect(translationSpy).toHaveBeenCalledWith(
      mockProps.description,
      `${mockProps.translation}.DESCRIPTION`,
    );
  });

  it('toggles the sidebar', () => {
    const wrapper = shallow(<SimpleUiWrapper title="mock title" />);
    expect(wrapper.state().showSidebar).toBe(false);
    wrapper.instance().toggleSidebar();
    expect(wrapper.state().showSidebar).toBe(true);
  });
});
