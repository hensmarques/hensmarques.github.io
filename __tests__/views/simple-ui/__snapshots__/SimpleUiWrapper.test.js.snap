// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimpleUiWrapper/> renders and matches the snapshot 1`] = `
<div
  className="simple-ui-container"
>
  <div
    className="simple-ui"
  >
    <div
      className="sidebar-container "
    >
      <Connect(ErrorBoundary)>
        <withRouter(Connect(Sidebar))
          toggleSidebar={[Function]}
        />
      </Connect(ErrorBoundary)>
    </div>
    <div
      className="main-container null"
    >
      <Connect(ErrorBoundary)>
        <Connect(SimpleUiHeader)
          showSidebar={false}
          toggleSidebar={[Function]}
        />
      </Connect(ErrorBoundary)>
      <div
        className="content-container"
      >
        <div
          className="page-heading"
        >
          <h2 />
        </div>
        <div
          className="description"
        >
          <p />
        </div>
        <div
          className="content"
        >
          <Connect(ErrorBoundary) />
        </div>
      </div>
    </div>
  </div>
</div>
`;
